src/lib/utils/options/declaration.ts

```diff
@@ -46,7 +46,7 @@ export class OptionDeclaration {
 
     scope: ParameterScope;
 
-    map: Object;
+    protected map: Object | Map<string, string> | 'object';
```

Comment:
`map` is really an internal implementation detail of what OptionDeclaration does and shouldn't be public.  If `map` remains `public` then anyone who relies on `map` will silently fail; because `Map` is also an `Object` TypeScript will not throw an error about incompatible types and `map` will look like an empty object. Changing `map` to `protected` ensures anyone who wants to continue to use `map` will need to extend this class and be aware of the change. Overall, we are better off making this breaking change and preventing silent failures for users.